### PR TITLE
[NEW] Create Team

### PR DESCRIPTION
**Pull request by gerzonc**
## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/lib/methods/loadMessagesForRoom.js [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+}) {
+	let params = { roomId: roomId || team.roomId, count: 50 };
+	let apiType;
 	if (latest) {
 		params = { ...params, latest: new Date(latest).toISOString() };
 	}
 
-	const apiType = this.roomTypeToApiType(t);
+	if (team.type) {
+		apiType = this.roomTypeToApiType('p');
```
Comment (diegolmello) [inline app/lib/methods/loadMessagesForRoom.js:14]: I'm not sure I got this.
Teams can be private and public.

File: app/sagas/createChannel.js [hunk 2]
```diff
@@ -76,12 +94,12 @@ const handleSuccess = function* handleSuccess({ data }) {
 	if (isMasterDetail) {
 		Navigation.navigate('DrawerNavigator');
 	}
-	goRoom({ item: data, isMasterDetail });
+	goRoom({ item: data.team ? data.team : data, isMasterDetail });
```
Comment (diegolmello) [inline app/sagas/createChannel.js:97]: It looks like you could call success with `data.team`

File: app/utils/goRoom.js [hunk 3]
```diff
@@ -9,9 +9,9 @@ const navigate = ({ item, isMasterDetail, ...props }) => {
 	}
 
 	navigationMethod('RoomView', {
-		rid: item.rid,
+		rid: item.roomId || item.rid,
 		name: RocketChat.getRoomTitle(item),
-		t: item.t,
+		t: item.type ? 'p' : item.t,
```
Comment (diegolmello) [inline app/utils/goRoom.js:14]: Why?

File: app/utils/goRoom.js [hunk 4]
```diff
@@ -9,9 +9,9 @@ const navigate = ({ item, isMasterDetail, ...props }) => {
 	}
 
 	navigationMethod('RoomView', {
-		rid: item.rid,
+		rid: item.roomId || item.rid,
```
Comment (diegolmello) [inline app/utils/goRoom.js:12]: Can't you call `goRoom` with rid?

File: app/views/CreateChannelView.js [hunk 5]
```diff
@@ -69,11 +69,12 @@ const styles = StyleSheet.create({
 
 class CreateChannelView extends React.Component {
 	static navigationOptions = () => ({
-		title: I18n.t('Create_Channel')
-	});
+		title: this.isTeam ? I18n.t('Create_Team') : I18n.t('Create_Channel')
```
Comment (diegolmello) [inline app/views/CreateChannelView.js:72]: It doesn't look correct.
How are you accessing `this` inside a `static`?
Maybe a `this.setHeader` would be better.

File: app/views/CreateChannelView.js [hunk 6]
```diff
@@ -89,12 +90,17 @@ class CreateChannelView extends React.Component {
...
-	state = {
-		channelName: '',
-		type: true,
-		readOnly: false,
-		encrypted: false,
-		broadcast: false
+	constructor(props) {
+		super(props);
+		const { route } = this.props;
+		this.isTeam = route?.params?.isTeam || false;
```
Comment (diegolmello) [inline app/views/CreateChannelView.js:96]: Can you use as state?

File: app/views/CreateChannelView.js [hunk 7]
```diff
@@ -312,18 +325,18 @@ class CreateChannelView extends React.Component {
 				keyboardVerticalOffset={128}
 			>
 				<StatusBar />
-				<SafeAreaView testID='create-channel-view'>
+				<SafeAreaView testID={this.isTeam ? 'create-team-view' : 'create-channel-view'}>
```
Comment (diegolmello) [inline app/views/CreateChannelView.js:328]: No need. The screen is the same.

File: app/views/CreateChannelView.js [hunk 8]
```diff
@@ -312,18 +325,18 @@ class CreateChannelView extends React.Component {
...
 								style={[styles.input, { backgroundColor: themes[theme].backgroundColor }]}
-								label={I18n.t('Channel_Name')}
+								label={this.isTeam ? I18n.t('Team_Name') : I18n.t('Channel_Name')}
 								value={channelName}
 								onChangeText={this.onChangeText}
-								placeholder={I18n.t('Channel_Name')}
+								placeholder={this.isTeam ? I18n.t('Team_Name') : I18n.t('Channel_Name')}
 								returnKeyType='done'
-								testID='create-channel-name'
+								testID={this.isTeam ? 'create-team-name' : 'create-channel-name'}
```
Comment (diegolmello) [inline app/views/CreateChannelView.js:339]: No reason to overcomplicate

File: e2e/helpers/data_setup.js [hunk 9]
```diff
@@ -57,6 +57,26 @@ const createChannelIfNotExists = async (channelname) => {
     }
 }
 
+const createTeamIfNotExists = async (teamname) => {
+    console.log(`Creating private team ${teamname}`)
+    try {
+        const team = await rocketchat.post('teams.create', {
+            "name": teamname,
+            "type": 1
```
Comment (diegolmello) [inline e2e/helpers/data_setup.js:65]: Is this a proper type?

File: e2e/helpers/data_setup.js [hunk 10]
```diff
@@ -57,6 +57,26 @@ const createChannelIfNotExists = async (channelname) => {
...
+        })
+        return team
+    } catch (createError) {
+        try { //Maybe it exists already?
+            const team = rocketchat.get(`teams.info?teamName=${teamname}`)
```
Comment (diegolmello) [inline e2e/helpers/data_setup.js:70]: You're not really using the return value.
I think you should remove it.

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/utils/goRoom.js (hunk 4 above)
Comment (gerzonc) [reply app/utils/goRoom.js:12]: API response for `teams.create` does not bring rid, it brings roomId instead.

File: app/utils/goRoom.js (hunk 3 above)
Comment (gerzonc) [reply app/utils/goRoom.js:14]: API response for `teams.create` does not bring a type, it returns a boolean value (e.g. `type : true`).

File: app/utils/goRoom.js (hunk 4 above)
Comment (diegolmello) [reply app/utils/goRoom.js:12]: `roomId` === `rid`. They are the same thing.

**Review by diegolmello — APPROVED** [review]
LGTM 🎉

**Pull request merged**